<a href="https://colab.research.google.com/github/michaelsteven1299/proyecto_michael-/blob/main/src/03_EDA.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Universidad Libre - Seccional Cali<br>Facultad de Ingeniería - Diplomado en Ciencia de Datos<br>(ↄ) Diego Fernando Marin, 2024

# 03_EDA
Plantilla para el desarrollo del proyecto del diplomado de Ciencia de Datos, aplicando buenas prácticas.

---

Este cuaderno se centra en comprender profundamente los datos mediante análisis estadístico y visualización. El EDA nos permite descubrir patrones, relaciones y características importantes en los datos.

**Propósito:** Obtener insights sobre los datos y preparar el terreno para la selección de características relevantes para el modelado.

**Tareas habituales:**
- Análisis univariante de variables
- Análisis de correlaciones
- Visualización de distribuciones
- Detección de patrones y tendencias
- Análisis de series temporales si aplica
- Identificación de relaciones entre variables
- Generación de hipótesis
- Documentación de insights importantes

**Cada sección debe tener**:
- La Pregunta que se realiza sobre los datos
- El código que genera las tablas o los gráficos donde encontramos la respuesta
- El código que da respuesta a dicha pregunta
- La interpretar del resultado
- Las conclusiones

Análisis Exploratorio

IMPORTAMOS LA TABLA DESDE TRUSTED

In [2]:
from google.colab import drive
drive.mount('/content/drive')

import pandas as pd
import numpy as np

ruta = '/content/drive/MyDrive/proyecto_oro/data/trusted/oro_dxy_trusted.csv'
df_oro = pd.read_csv(ruta)

print("Dataset cargado")
print(f"Dimensiones: {df_oro.shape}")
df_oro.head()

Mounted at /content/drive
Dataset cargado
Dimensiones: (1432, 47)


,DATE,gvz_Close,gvz_High,gvz_Low,gvz_Open,cobre_Close,cobre_High,cobre_Low,cobre_Open,cobre_Volume,eur_usd_Close,eur_usd_High,eur_usd_Low,eur_usd_Open,bono_10y_Close,bono_10y_High,bono_10y_Low,bono_10y_Open,plata_xagusd_Close,plata_xagusd_High,plata_xagusd_Low,plata_xagusd_Open,plata_xagusd_Volume,dxy_Close,dxy_High,dxy_Low,dxy_Open,vix_Close,vix_High,vix_Low,vix_Open,wti_crudo_Close,wti_crudo_High,wti_crudo_Low,wti_crudo_Open,wti_crudo_Volume,oro_xauusd_Close,oro_xauusd_High,oro_xauusd_Low,oro_xauusd_Open,oro_xauusd_Volume,yield_real_10y,breakeven_inflacion,spread_curva_10y2y,bono_2y,target_clasificacion,target_retorno
0,2021-01-04,21.860001,22.360001,21.240000,22.090000,3.5530,3.6005,3.5455,3.5590,864.0,1.225070,1.230999,1.217137,1.224905,0.917,0.953,0.907,0.935,25.35,25.610001,24.870001,25.540001,54026000.0,89.879997,89.940002,89.419998,89.930000,26.969999,29.190001,22.559999,23.040001,47.619999,49.830002,47.180000,48.400002,528525.0,182.330002,182.399994,180.960007,181.970001,14331400.0,-1.08,2.01,0.82,0.11,1,0.002962
1,2021-01-05,22.139999,22.260000,20.920000,21.110001,3.6405,3.6555,3.5900,3.5900,512.0,1.225160,1.229483,1.224995,1.225295,0.955,0.963,0.927,0.937,25.67,25.740000,25.280001,25.459999,32882000.0,89.440002,89.900002,89.430000,89.900002,25.340000,28.600000,24.799999,26.940001,49.930000,50.200001,47.240002,47.380001,643191.0,182.869995,183.210007,181.820007,182.869995,12718800.0,-1.07,2.03,0.83,0.13,0,-0.016241
2,2021-01-06,21.700001,21.809999,20.590000,20.860001,3.6500,3.6905,3.6400,3.6505,842.0,1.230027,1.235025,1.226693,1.229861,1.042,1.054,1.000,1.000,25.34,25.400000,24.719999,25.340000,54978700.0,89.529999,89.800003,89.209999,89.480003,25.070000,26.770000,22.139999,25.480000,50.630001,50.939999,49.480000,49.820000,509365.0,179.899994,181.580002,178.240005,181.490005,18453500.0,-1.02,2.06,0.90,0.14,0,-0.002335
3,2021-01-07,20.660000,21.420000,20.389999,21.240000,3.6955,3.7080,3.6430,3.6590,655.0,1.234111,1.234568,1.224665,1.233776,1.071,1.088,1.054,1.056,25.26,25.370001,25.049999,25.219999,27335800.0,89.830002,89.970001,89.320000,89.320000,22.370001,23.910000,22.250000,23.670000,50.830002,51.279999,50.389999,50.529999,369292.0,179.479996,179.919998,178.839996,179.690002,7110200.0,-1.01,2.09,0.94,0.14,0,-0.034210
4,2021-01-08,22.160000,23.469999,19.540001,20.120001,3.6735,3.7200,3.6400,3.7165,782.0,1.227144,1.228215,1.221493,1.226873,1.105,1.126,1.075,1.088,23.51,24.760000,22.740000,24.730000,107226800.0,90.099998,90.250000,89.660004,89.830002,21.559999,23.340000,21.420000,22.430000,52.240002,52.750000,50.810001,50.930000,499416.0,173.339996,176.990005,171.479996,176.830002,24399900.0,-0.93,2.06,0.99,0.14,0,-0.001961


tipo de datos

In [3]:
df_oro.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1432 entries, 0 to 1431
Data columns (total 47 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   DATE                  1432 non-null   object 
 1   gvz_Close             1432 non-null   float64
 2   gvz_High              1432 non-null   float64
 3   gvz_Low               1432 non-null   float64
 4   gvz_Open              1432 non-null   float64
 5   cobre_Close           1432 non-null   float64
 6   cobre_High            1432 non-null   float64
 7   cobre_Low             1432 non-null   float64
 8   cobre_Open            1432 non-null   float64
 9   cobre_Volume          1432 non-null   float64
 10  eur_usd_Close         1432 non-null   float64
 11  eur_usd_High          1432 non-null   float64
 12  eur_usd_Low           1432 non-null   float64
 13  eur_usd_Open          1432 non-null   float64
 14  bono_10y_Close        1432 non-null   float64
 15  bono_10y_High        

contruimos variables rezagados por 30 dias con el precio del cierre del oro

In [4]:
df_oro['oro_return_1d'] = df_oro['oro_xauusd_Close'].pct_change()

for lag in range(1, 31):
    df_oro[f'oro_lag_{lag}'] = df_oro['oro_xauusd_Close'].shift(lag)

print(f"Shape del dataset ahora: {df_oro.shape}")
df_oro.head()

Shape del dataset ahora: (1432, 78)


,DATE,gvz_Close,gvz_High,gvz_Low,gvz_Open,cobre_Close,cobre_High,cobre_Low,cobre_Open,cobre_Volume,eur_usd_Close,eur_usd_High,eur_usd_Low,eur_usd_Open,bono_10y_Close,bono_10y_High,bono_10y_Low,bono_10y_Open,plata_xagusd_Close,plata_xagusd_High,plata_xagusd_Low,plata_xagusd_Open,plata_xagusd_Volume,dxy_Close,dxy_High,dxy_Low,dxy_Open,vix_Close,vix_High,vix_Low,vix_Open,wti_crudo_Close,wti_crudo_High,wti_crudo_Low,wti_crudo_Open,wti_crudo_Volume,oro_xauusd_Close,oro_xauusd_High,oro_xauusd_Low,oro_xauusd_Open,oro_xauusd_Volume,yield_real_10y,breakeven_inflacion,spread_curva_10y2y,bono_2y,target_clasificacion,target_retorno,oro_return_1d,oro_lag_1,oro_lag_2,oro_lag_3,oro_lag_4,oro_lag_5,oro_lag_6,oro_lag_7,oro_lag_8,oro_lag_9,oro_lag_10,oro_lag_11,oro_lag_12,oro_lag_13,oro_lag_14,oro_lag_15,oro_lag_16,oro_lag_17,oro_lag_18,oro_lag_19,oro_lag_20,oro_lag_21,oro_lag_22,oro_lag_23,oro_lag_24,oro_lag_25,oro_lag_26,oro_lag_27,oro_lag_28,oro_lag_29,oro_lag_30
0,2021-01-04,21.860001,22.360001,21.240000,22.090000,3.5530,3.6005,3.5455,3.5590,864.0,1.225070,1.230999,1.217137,1.224905,0.917,0.953,0.907,0.935,25.35,25.610001,24.870001,25.540001,54026000.0,89.879997,89.940002,89.419998,89.930000,26.969999,29.190001,22.559999,23.040001,47.619999,49.830002,47.180000,48.400002,528525.0,182.330002,182.399994,180.960007,181.970001,14331400.0,-1.08,2.01,0.82,0.11,1,0.002962,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1,2021-01-05,22.139999,22.260000,20.920000,21.110001,3.6405,3.6555,3.5900,3.5900,512.0,1.225160,1.229483,1.224995,1.225295,0.955,0.963,0.927,0.937,25.67,25.740000,25.280001,25.459999,32882000.0,89.440002,89.900002,89.430000,89.900002,25.340000,28.600000,24.799999,26.940001,49.930000,50.200001,47.240002,47.380001,643191.0,182.869995,183.210007,181.820007,182.869995,12718800.0,-1.07,2.03,0.83,0.13,0,-0.016241,0.002962,182.330002,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2,2021-01-06,21.700001,21.809999,20.590000,20.860001,3.6500,3.6905,3.6400,3.6505,842.0,1.230027,1.235025,1.226693,1.229861,1.042,1.054,1.000,1.000,25.34,25.400000,24.719999,25.340000,54978700.0,89.529999,89.800003,89.209999,89.480003,25.070000,26.770000,22.139999,25.480000,50.630001,50.939999,49.480000,49.820000,509365.0,179.899994,181.580002,178.240005,181.490005,18453500.0,-1.02,2.06,0.90,0.14,0,-0.002335,-0.016241,182.869995,182.330002,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3,2021-01-07,20.660000,21.420000,20.389999,21.240000,3.6955,3.7080,3.6430,3.6590,655.0,1.234111,1.234568,1.224665,1.233776,1.071,1.088,1.054,1.056,25.26,25.370001,25.049999,25.219999,27335800.0,89.830002,89.970001,89.320000,89.320000,22.370001,23.910000,22.250000,23.670000,50.830002,51.279999,50.389999,50.529999,369292.0,179.479996,179.919998,178.839996,179.690002,7110200.0,-1.01,2.09,0.94,0.14,0,-0.034210,-0.002335,179.899994,182.869995,182.330002,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
4,2021-01-08,22.160000,23.469999,19.540001,20.120001,3.6735,3.7200,3.6400,3.7165,782.0,1.227144,1.228215,1.221493,1.226873,1.105,1.126,1.075,1.088,23.51,24.760000,22.740000,24.730000,107226800.0,90.099998,90.250000,89.660004,89.830002,21.559999,23.340000,21.420000,22.430000,52.240002,52.750000,50.810001,50.930000,499416.0,173.339996,176.990005,171.479996,176.830002,24399900.0,-0.93,2.06,0.99,0.14,0,-0.001961,-0.034210,179.479996,179.899994,182.869995,182.330002,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


alisis decriptivo para todas las variables

In [5]:
pd.set_option('display.max_rows', None)
pd.set_option('display.max_columns', None)

df_oro.describe()

,gvz_Close,gvz_High,gvz_Low,gvz_Open,cobre_Close,cobre_High,cobre_Low,cobre_Open,cobre_Volume,eur_usd_Close,eur_usd_High,eur_usd_Low,eur_usd_Open,bono_10y_Close,bono_10y_High,bono_10y_Low,bono_10y_Open,plata_xagusd_Close,plata_xagusd_High,plata_xagusd_Low,plata_xagusd_Open,plata_xagusd_Volume,dxy_Close,dxy_High,dxy_Low,dxy_Open,vix_Close,vix_High,vix_Low,vix_Open,wti_crudo_Close,wti_crudo_High,wti_crudo_Low,wti_crudo_Open,wti_crudo_Volume,oro_xauusd_Close,oro_xauusd_High,oro_xauusd_Low,oro_xauusd_Open,oro_xauusd_Volume,yield_real_10y,breakeven_inflacion,spread_curva_10y2y,bono_2y,target_clasificacion,target_retorno,oro_return_1d,oro_lag_1,oro_lag_2,oro_lag_3,oro_lag_4,oro_lag_5,oro_lag_6,oro_lag_7,oro_lag_8,oro_lag_9,oro_lag_10,oro_lag_11,oro_lag_12,oro_lag_13,oro_lag_14,oro_lag_15,oro_lag_16,oro_lag_17,oro_lag_18,oro_lag_19,oro_lag_20,oro_lag_21,oro_lag_22,oro_lag_23,oro_lag_24,oro_lag_25,oro_lag_26,oro_lag_27,oro_lag_28,oro_lag_29,oro_lag_30
count,1432.000000,1432.000000,1432.000000,1432.000000,1432.000000,1432.000000,1432.000000,1432.000000,1432.000000,1432.000000,1432.000000,1432.000000,1432.000000,1432.000000,1432.000000,1432.000000,1432.000000,1432.000000,1432.000000,1432.000000,1432.000000,1.432000e+03,1432.000000,1432.000000,1432.000000,1432.000000,1432.000000,1432.000000,1432.000000,1432.000000,1432.000000,1432.000000,1432.000000,1432.000000,1.432000e+03,1432.000000,1432.000000,1432.000000,1432.000000,1.432000e+03,1432.000000,1432.000000,1432.000000,1432.000000,1432.000000,1432.000000,1431.000000,1431.000000,1430.000000,1429.000000,1428.000000,1427.000000,1426.000000,1425.000000,1424.000000,1423.000000,1422.000000,1421.000000,1420.000000,1419.000000,1418.000000,1417.000000,1416.000000,1415.000000,1414.000000,1413.000000,1412.000000,1411.000000,1410.000000,1409.000000,1408.000000,1407.000000,1406.000000,1405.000000,1404.000000,1403.000000,1402.000000
mean,18.255803,18.881208,17.735154,18.275398,4.387134,4.417425,4.351806,4.384465,1488.518855,1.111615,1.114918,1.108184,1.111607,3.456784,3.488817,3.422476,3.454431,29.540978,29.856732,29.178939,29.547786,2.747261e+07,100.805726,101.116627,100.484253,100.791264,19.297151,20.613764,18.496830,19.533568,76.688066,78.120056,75.231983,76.726683,3.225746e+05,230.837940,231.987144,229.531006,230.852870,8.902917e+06,1.107228,2.352046,0.199895,3.259378,0.513268,0.000554,0.000554,230.741824,230.645433,230.545374,230.448095,230.353160,230.250098,230.141789,230.031552,229.921159,229.809571,229.691471,229.573965,229.463347,229.352729,229.250240,229.136165,229.017343,228.899080,228.770014,228.643173,228.513260,228.383652,228.249702,228.118650,227.990455,227.858158,227.725801,227.593255,227.458261,227.322782
std,5.064775,5.327968,4.807618,5.087157,0.680156,0.686203,0.670818,0.678018,5005.819220,0.057933,0.057389,0.058306,0.057922,1.129767,1.132441,1.127103,1.129932,15.112067,15.444606,14.681266,15.126268,2.841016e+07,5.225186,5.287254,5.154959,5.230255,5.154051,5.950985,4.619353,5.235165,13.467718,14.091644,12.859000,13.514650,1.299677e+05,86.031696,86.814179,85.018536,86.104132,5.914727e+06,1.172702,0.151761,0.650266,1.601825,0.499999,0.011142,0.011146,85.984824,85.937530,85.884246,85.835542,85.790606,85.732286,85.664734,85.593659,85.522282,85.448676,85.362583,85.277613,85.205705,85.133777,85.076278,84.997858,84.910181,84.823565,84.714588,84.610272,84.499276,84.388840,84.268718,84.154904,84.047184,83.930408,83.813378,83.695806,83.572606,83.448163
min,10.220000,10.500000,8.710000,10.390000,3.210500,3.254000,3.150000,3.154000,0.000000,0.959619,0.967006,0.954016,0.959619,0.917000,0.953000,0.907000,0.935000,16.379999,16.480000,16.190001,16.389999,3.967700e+06,89.440002,89.800003,89.209999,89.320000,11.860000,12.230000,10.620000,11.530000,47.619999,49.830002,47.180000,47.380001,0.000000e+00,151.229996,151.960007,150.570007,150.699997,1.436500e+06,-1.190000,2.010000,-1.080000,0.090000,0.000000,-0.102742,-0.102742,151.229996,151.229996,151.229996,151.229996,151.229996,151.229996,151.229996,151



El dataset cubre 1432 días de trading desde enero de 2021 hasta mediados de 2026, sin valores nulos. El oro (oro_xauusd_Close) tiene una media de 230.84 dólares con una desviación estándar de 86.03, lo que refleja la fuerte tendencia de crecimiento del periodo: pasó de un mínimo de 151.23 a un máximo de 495.90 dólares, más que triplicando su valor. Las variables de volumen muestran mucha dispersión respecto a su media, lo que sugiere posibles valores atípicos en días de actividad inusual. Las tasas de interés (bono_10y_Close, bono_2y) reflejan el ciclo de subida de tasas de la Reserva Federal, pasando de niveles cercanos a cero hasta casi 5%. Los lags del oro tienen estadísticos muy similares a la variable original, como era de esperarse, con una leve reducción en el conteo de datos a medida que aumenta el número de días rezagados. Por último, target_retorno tiene una media casi nula (0.055%) y una desviación estándar de 1.11%, con un rango que va de -10.27% a +6.36%, mostrando que el movimiento diario típico es pequeño pero con días extremos que conviene revisar más adelante.

